components/script/dom/htmlformelement.rs

```diff
@@ -417,55 +409,77 @@ impl HTMLFormElement {
 
         let mut load_data = LoadData::new(action_components, doc.get_referrer_policy(), Some(doc.url().clone()));
 
-        let parsed_data = match enctype {
-            FormEncType::UrlEncoded => {
-                load_data.headers.set(ContentType::form_url_encoded());
-
-                form_urlencoded::Serializer::new(String::new())
-                    .encoding_override(Some(self.pick_encoding()))
-                    .extend_pairs(form_data.into_iter().map(|field| (field.name.clone(), field.value_str())))
-                    .finish()
-            }
-            FormEncType::FormDataEncoded => {
-                let boundary = self.generate_boundary();
-                let mime = mime!(Multipart / FormData; Boundary =(&boundary));
-                load_data.headers.set(ContentType(mime));
-
-                self.encode_multipart_form_data(&mut form_data, None, boundary)
-            }
-            FormEncType::TextPlainEncoded => {
-                load_data.headers.set(ContentType(mime!(Text / Plain)));
-
-                self.encode_plaintext(&mut form_data)
-            }
-        };
-
         // Step 18
         let win = window_from_node(self);
         match (&*scheme, method) {
-            // https://html.spec.whatwg.org/multipage/#submit-dialog
-            (_, FormMethod::FormDialog) => return, // Unimplemented
+            (_, FormMethod::FormDialog) => {
+                // TODO: Submit dialog
+                // https://html.spec.whatwg.org/multipage/#submit-dialog
+            }
             // https://html.spec.whatwg.org/multipage/#submit-mutate-action
-            ("http", FormMethod::FormGet) | ("https", FormMethod::FormGet) => {
-                // FIXME(SimonSapin): use url.query_pairs_mut() here.
-                load_data.url.set_query(Some(&*parsed_data));
+            ("http", FormMethod::FormGet) | ("https", FormMethod::FormGet) | ("data", FormMethod::FormGet) => {
+                load_data.headers.set(ContentType::form_url_encoded());
+                self.mutate_action_url(&mut form_data, &mut load_data.url);
                 self.plan_to_navigate(load_data, &win);
```

Comment:
@KiChjang it will navigate here, and well, I admit that your suggestion is right. I will make it more coherent with spec.